forte/integrals/conventional_integrals.cc

```diff
@@ -165,6 +167,75 @@ void ConventionalIntegrals::set_tei(size_t p, size_t q, size_t r, size_t s, doub
         aphys_tei_bb_[index] = value;
 }
 
+void ConventionalIntegrals::set_tei_from_asints(std::shared_ptr<ActiveSpaceIntegrals> as_ints,
+                                                bool alpha1, bool alpha2) {
+    for (size_t p = 0; p < ncmo_; ++p) {
+        for (size_t q = 0; q < ncmo_; ++q) {
+            for (size_t r = 0; r < ncmo_; ++r) {
+                for (size_t s = 0; s < ncmo_; ++s) {
+                    size_t index = aptei_index(p, q, r, s);
+                    if (alpha1 == true and alpha2 == true)
+                        aphys_tei_aa_[index] = as_ints->tei_aa(p, q, r, s);
+                    if (alpha1 == true and alpha2 == false)
+                        aphys_tei_ab_[index] = as_ints->tei_ab(p, q, r, s);
+                    if (alpha1 == false and alpha2 == false)
+                        aphys_tei_bb_[index] = as_ints->tei_bb(p, q, r, s);
+                }
+            }
+        }
+    }
+}
+
+void ConventionalIntegrals::build_from_asints(std::shared_ptr<ActiveSpaceIntegrals> as_ints) {
+    outfile->Printf("\n  Updating one-electron integrals from Hbar");
+    set_oei_from_asints(as_ints, true);
+    set_oei_from_asints(as_ints, false);
+    outfile->Printf("\n  Updating two-electron integrals from Hbar");
+    set_tei_from_asints(as_ints, true, true);
+    set_tei_from_asints(as_ints, true, false);
+    set_tei_from_asints(as_ints, false, false);
+}
+
+void ConventionalIntegrals::make_fock_matrix_from_value(std::shared_ptr<psi::Matrix> gamma_a,
+                                                        std::shared_ptr<psi::Matrix> gamma_b) {
+    psi::SharedMatrix Fa(new psi::Matrix("Fa_fill", ncmo_, ncmo_));
```

Comment:
This does not perform exactly the same as the JK version. The JK version contains the frozen orbital contributions to the Fock.

Reply:
Because this is a general function in the integral class but it performs differently than the default one. I think we should either throw an error or print a warning for what is not intended to do.

A potential issue is that in cases where we need to semicanonicalize both frozen-core and core orbitals, this part of the code will generate an indirect bug.